File: leetcode/tree/binary_tree_right_side_view.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func rightSideView(root *TreeNode) []int {
	if root == nil {
		return nil
	}
	var q []*TreeNode
	q = append(q, root)
	var res []int
	for len(q) != 0 {
		var inner []int
		size := len(q)
		for i := 0; i < size; i++ {
			tmp := q[0]
			q = q[1:]
			if tmp.Left != nil {
				q = append(q, tmp.Left)
			}
			if tmp.Right != nil {
				q = append(q, tmp.Right)
			}
			inner = append(inner, tmp.Val)
		}
		res = append(res, inner[len(inner)-1])
	}
	return res
}
func main() {
	root := &TreeNode{3, nil, nil}
	root.Left = &TreeNode{9, nil, nil}
	root.Right = &TreeNode{20, nil, nil}
	root.Right.Left = &TreeNode{15, nil, nil}
	root.Right.Right = &TreeNode{7, nil, nil}
	out := rightSideView(root)
	fmt.Println(out)
}

File: leetcode/tree/binary_tree_right_side_view.py
from common import TreeNode
from typing import List

class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        arr = []
        def helper(root):
            if not root:
                return
            q = [root]
            while len(q) != 0:
                inner = []
                size = len(q)
                for i in range(len(q)):
                    tmp = q.pop(0)
                    if tmp.left:
                        q.append(tmp.left)
                    if tmp.right:
                        q.append(tmp.right)
                    inner.append(tmp.val)
                arr.append(inner[-1])
        helper(root)
        return arr

if __name__ == "__main__":
    root = TreeNode(3)
    root.left = TreeNode(9)
    root.right = TreeNode(20)
    root.right.left = TreeNode(15)
    root.right.right = TreeNode(7)
    print(Solution().rightSideView(root))